budgetportal/tests/test_prov_infra_projects.py

```diff
@@ -1,247 +1,959 @@
+        search_button = selenium.find_element_by_xpath('//*[@id="Search-Button"]')
+        search_field.send_keys(province)
+        search_button.click()
+        time.sleep(self.timeout)
+        filtered_num_of_projects = selenium.find_element_by_xpath(
+            '//*[@id="num-matching-projects-field"]'
+        ).text
+        filtered_num_of_projects = int(filtered_num_of_projects)
+        self.assertEqual(filtered_num_of_projects, 5)
+
+
+class ProvInfraProjectAPIDepartmentTestCase(APITransactionTestCase):
+    def setUp(self):
+        self.url = reverse("provincial-infrastructure-project-api-list")
+        self.facet_url = reverse("provincial-infrastructure-project-api-facets")
+        self.fin_year = FinancialYear.objects.create(slug="2030-31")
+        self.quarter = Quarter.objects.create(number=1)
+        self.date = date(year=2050, month=1, day=1)
+        self.irm_snapshot = IRMSnapshot.objects.create(
+            financial_year=self.fin_year, quarter=self.quarter, date_taken=self.date
+        )
+        self.project_1 = ProvInfraProject.objects.create(IRM_project_id=1)
+        ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot,
+            project=self.project_1,
+            department="Department 1",
+            province="Eastern Cape",
+            estimated_completion_date=self.date
+        )
+        self.project_2 = ProvInfraProject.objects.create(IRM_project_id=2)
+        ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot,
+            project=self.project_2,
+            department="Department 1",
+            province="Free State",
+            estimated_completion_date=self.date
+        )
+        self.project_3 = ProvInfraProject.objects.create(IRM_project_id=3)
+        ProvInfraProjectSnapshot.objects.create(
+            irm_snapshot=self.irm_snapshot,
+            project=self.project_3,
+            department="Department 2",
+            province="Eastern Cape",
+            estimated_completion_date=self.date
+        )
+        ProvInfraProjectIndex().reindex()
 
-        number_of_results = len(response.data["results"])
-        self.assertLessEqual(number_of_results, 20)
+    def tearDown(self):
+        ProvInfraProjectIndex().clear()
 
     def test_filter_by_department(self):
         department = "Department 1"
-        project = ProvInfraProject.objects.get(department=department)
-
         data = {"department": department}
+
         response = self.client.get(self.url, data)
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
-        number_of_projects = len(response.data["results"])
-        response_data = response.data["results"][0]
-        self.assertEqual(number_of_projects, 1)
-        self.assertEqual(response_data["name"], project.name)
+        number_of_projects = response.data["count"]
+        self.assertEqual(number_of_projects, 2)
```

Comment:
You don't need to change this now, but I'm just thinking, this could be as short as

```
response = self.client.get(self.url, {"department": department})
self.assertEqual(response.status_code, status.HTTP_200_OK)
self.assertEqual(2, response.data["count"])
```

You don't have to bind everything to a variable :)